app/routes.py

```diff
@@ -1,2 +1,193 @@
-from flask import Blueprint
+from app import db
+from flask import Blueprint, request, abort, jsonify, make_response
+from datetime import datetime
+from dotenv import load_dotenv
+import os
+import requests
+from app.models.task import Task
+from app.models.goal import Goal
+from app.utils.route_wrappers import require_instance_or_404
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    """
+    Retrieve all tasks. Allows for use of query parameters.
+    Returns JSON list of task dictionaries. """
+    query = Task.query # Base query
+
+    # Query params, adding to query where indicated
+    sort = request.args.get("sort")
+    if sort == "asc":
+        query = query.order_by(Task.title)
+    elif sort == "desc":
+        query = query.order_by(Task.title.desc())
+    
+    query = query.all() # Final query
+    
+    # Returns jsonified list of task dicionaries
+    return jsonify([task.to_dict() for task in query]), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+@require_instance_or_404
+def get_task(task):
+    """Retrieve one stored task by id."""
+    if task.goal_id:
```

Comment:
This works but to simplify the code I would recommend pushing this check to the to_dict method and add the goal_id key-value pair if goal_id exists inside the to_dict function.